### PR TITLE
Added SQL99 compliant table columns

### PR DESCRIPTION
**Pull request by Autophagy**
Adds 4 new columns:
- **table_catalog**: returns the name of the database (always "CrateDB")
- **table_type**: returns the type of the table (always 'BASE TABLE', as we dont implement 'VIEW' (for a View), 'LOCAL TEMPORARY' (for a local temporary Table) or 'GLOBAL TEMPORARY' (for a global temporary Table)
- **self_referencing_column_name**: unimplemented, always null
- **reference_generation**: unimplemented, always null

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java [hunk 1]
```diff
@@ -45,6 +45,22 @@ public BytesRef value() {
...
+        public BytesRef value() {
+            return new BytesRef("CrateDB");
+        }
+    }
+
+    public static class PartitionsTableTypeExpression extends InformationTablePartitionsExpression<BytesRef> {
+
+        @Override
+        public BytesRef value() {
+            return new BytesRef("BASE TABLE");
```
Comment (chaudum) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: you can use a static BytesRef instance to minimize allocations

File: sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java [hunk 2]
```diff
@@ -30,6 +30,8 @@
 abstract class InformationTablePartitionsExpression<T>
     extends RowContextCollectorExpression<PartitionInfo, T> {
 
+    public static final BytesRef tableType = new BytesRef("BASE TABLE");
```
Comment (chaudum) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: IMO the table type should be defined as part of the TableInfo, @mfussenegger what do you think?
how many table types are there?
if that is too much overhead, I would define the static variable inside the expression https://github.com/crate/crate/pull/5641/files#diff-1f715bb8929ec4f8885636b46069d228R156

Comment (chaudum) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: ah, I see it's also used down there. https://github.com/crate/crate/pull/5641/files#diff-b98d4715cbfe1b1d080ebe91ea61756aR62

Comment (chaudum) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: anyway, naming: `TABLE_TYPE`

File: sql/src/main/java/io/crate/operation/reference/information/InformationSchemaExpressionFactories.java [hunk 3]
```diff
@@ -235,6 +257,22 @@ public Boolean value() {
                         return null;
                     }
                 })
+            .put(InformationTablesTableInfo.Columns.SELF_REFERENCING_COLUMN_NAME, () -> new
+                RowContextCollectorExpression<TableInfo, BytesRef>() {
+
+                @Override
+                public BytesRef value() {
+                    return null;
```
Comment (chaudum) [inline sql/src/main/java/io/crate/operation/reference/information/InformationSchemaExpressionFactories.java]: does it make sense to create a `RowCollectNullExpressionFactory` class that is reused?

File: sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java (hunk 2 above)
Comment (Autophagy) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: There are 4: 'BASE TABLE', 'VIEW', 'LOCAL TEMPORARY', 'GLOBAL TEMPORARY

File: sql/src/main/java/io/crate/operation/reference/information/InformationSchemaExpressionFactories.java (hunk 3 above)
Comment (Autophagy) [inline sql/src/main/java/io/crate/operation/reference/information/InformationSchemaExpressionFactories.java]: Yeah, it does - I believe @christianbader did something similar.

File: sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java (hunk 2 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/information/InformationTablePartitionsExpression.java]: Think it's okay to keep it hard-coded here for now. Wouldn't anticipate too much here. We might end up *not* using TableInfo if we add views for example. So I'd keep it simple for now as it's easy to change this way, but much harder if we add it to TableInfo and it ends up being wrong there.

**Pull request merged**